megamek/src/megamek/common/BattleArmor.java

```diff
@@ -528,6 +528,10 @@ && hasWorkingMisc(MiscType.F_PARTIAL_WING)
                     && ignoreGameLessThanThin) {
                 mp++;
             }
+        } else {
+            if ((mp > 0) && hasWorkingMisc(MiscType.F_PARTIAL_WING)) {
+                mp++;
+            }
         }
```

Comment:
can't that whole block starting at 522 be written as 
```
if ((mp > 0) && hasWorkingMisc(MiscType.F_PARTIAL_WING) && 
  (mpCalculationSetting.ignoreWeather || (game==null) || game.getPlanetaryConditions().getAtmosphere().isLighterThan(Atmosphere.THIN))) {
  mp++;
}
```
?

Reply:
I think that is harder to read and easier for some one to break things when updating something.

Reply:
I tend to agree with @kuronekochomusuke on this one, although I think adding a comment to clearly state which branch is needed for which project would be a good move.